File: crates/krilla/examples/empty_document.rs
//! This example contains the most basic document you can create: A document
//! with a single empty page.

use std::path;

use krilla::page::PageSettings;
use krilla::Document;

fn main() {
    // First, we create a new document. This represents a single PDF document.
    let mut document = Document::new();
    // We can now successively add new pages by calling `start_page`, or `start_page_with`
    // if we want to pass custom page settings.
    let page = document.start_page_with(PageSettings::from_wh(300.0, 600.0).unwrap());
    page.finish();

    // Create the PDF
    let pdf = document.finish().unwrap();

    let path = path::absolute("empty_document.pdf").unwrap();
    eprintln!("Saved PDF to '{}'", path.display());

    // Write the PDF to a file.
    std::fs::write(path, &pdf).unwrap();
}

File: crates/krilla-py/examples/empty_document.py
"""This example contains the most basic document you can create: A document
with a single empty page.
"""

from pathlib import Path

from krilla import Document, PageSettings


def main():
    # First, we create a new document. This represents a single PDF document.
    document = Document()
    # We can now successively add new pages by calling `start_page`, or
    # `start_page_with` if we want to pass custom page settings.
    with document.start_page_with(PageSettings.from_wh(300.0, 600.0)):
        pass

    # Create the PDF
    pdf = document.finish()

    path = Path("empty_document.pdf").absolute()
    print(f"Saved PDF to '{path}'")

    # Write the PDF to a file.
    with open(path, "wb") as f:
        f.write(pdf)


if __name__ == "__main__":
    main()
